src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -908,6 +908,44 @@ public static unsafe object IsInstanceOf(MethodTable* pTargetType, object obj)
                 return IsInstanceOfClass(pTargetType, obj);
         }
 
+        [RuntimeExport("RhTypeCast_IsInstanceOfException")]
+        public static unsafe bool IsInstanceOfException(MethodTable* pTargetType, object? obj)
+        {
+            // Based on IsInstanceOfClass_Helper
+
+            if (obj == null)
+                return false;
+
+            MethodTable* pObjType = obj.GetMethodTable();
+            if (pTargetType->IsCloned)
+                pTargetType = pTargetType->CanonicalEEType;
+
+            if (pObjType->IsCloned)
+                pObjType = pObjType->CanonicalEEType;
+
+            if (pObjType == pTargetType)
+                return true;
+
+            if (pTargetType->HasGenericVariance && pObjType->HasGenericVariance)
+                return CastCache.AreTypesAssignableInternal(pObjType, pTargetType, AssignmentVariation.BoxedSource, null);
+
+            if (pObjType->IsArray)
+                return false;
+
```

Comment:
```suggestion
            if (pObjType->IsArray)
            {
                // arrays can be cast to System.Object
                if (WellKnownEETypes.IsSystemObject(pTargetType))
                {
                    return obj;
                }

                // arrays can be cast to System.Array
                if (WellKnownEETypes.IsSystemArray(pTargetType))
                {
                    return obj;
                }

                return null;
            }

```

Reply:
Ah, I've just removed the variance piece and can't apply your patch 🙂  (also it's true/false return type)

Reply:
Why IsArray should be handled?

Reply:
ah, because it can be thrown/handled in IL as well, right?

Reply:
Right.

Reply:
One could use the same argument to keep the variance checks since one could throw delegates and they do have variance. But also, I don't care that much - NativeAOT does a failfast for anything that doesn't derive from System.Exception anyway.